File: src/com/l2kt/gameserver/model/actor/instance/HalishaChest.java
package com.l2kt.gameserver.model.actor.instance;

import com.l2kt.gameserver.model.actor.template.NpcTemplate;

public final class HalishaChest extends Monster
{
	public HalishaChest(int objectId, NpcTemplate template)
	{
		super(objectId, template);
		
		setNoRndWalk(true);
		setShowSummonAnimation(true);
		disableCoreAI(true);
	}
	
	@Override
	public boolean isMovementDisabled()
	{
		return true;
	}
	
	@Override
	public boolean hasRandomAnimation()
	{
		return false;
	}
}

File: src/com/l2kt/gameserver/model/actor/instance/HalishaChest.kt
package com.l2kt.gameserver.model.actor.instance

import com.l2kt.gameserver.model.actor.template.NpcTemplate

class HalishaChest(objectId: Int, template: NpcTemplate) : Monster(objectId, template) {
    init {

        setIsNoRndWalk(true)
        isShowSummonAnimation = true
        disableCoreAI(true)
    }

    override fun isMovementDisabled(): Boolean {
        return true
    }

    override fun hasRandomAnimation(): Boolean {
        return false
    }
}